src/tools/genpolicy/rules.rego

```diff
@@ -276,16 +276,15 @@ allow_by_sandbox_name(p_oci, i_oci, p_storages, i_storages, s_name) {
 
 allow_sandbox_name(p_s_name, i_s_name) {
     print("allow_sandbox_name 1: start")
-
+    not policy_data.is_generated_name
     p_s_name == i_s_name
 
     print("allow_sandbox_name 1: true")
 }
 allow_sandbox_name(p_s_name, i_s_name) {
     print("allow_sandbox_name 2: start")
-
-    # TODO: should generated names be handled differently?
-    contains(p_s_name, "$(generated-name)")
+    policy_data.is_generated_name
```

Comment:
Alternative idea: we could merge the two rules here if we assembled a regexp from either `name` or `generateName` in genpolicy. 

```rust
let p_s_name = if let Some(generateName) = obj.metadata.generateName {
  // https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-subdomain-names
  format!("^{}[a-z0-9.-]*[a-z0-9]$", regex::escape(generateName))
} else {
  format!("^{}$", regex::escape(obj.metadata.name))
}
```

Reply:
I really like this alternative since it makes the policy a bit smaller by merging the two rules, and we no longer need to pass `is_generated_name` through the policy data. Added. Thanks @burgerdev !